Frontend/src/components/Register.js

```diff
@@ -0,0 +1,245 @@
+import React, { useState } from "react";
+import { useForm } from 'react-hook-form';
+import {useHistory} from "react-router-dom";
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as Yup from 'yup';
+
+import AuthService from "../services/auth.service";
+
+const Register = () => {
+  const history = useHistory();
+  const [successful, setSuccessful] = useState(false);
+  const [message, setMessage] = useState("");
+
+  const validationSchema = Yup.object().shape({
+    name: Yup.string()
+        .required('Name is required')
+        .min(10, 'Name must be at least 10 characters')
+        .max(30, 'Name must not exceed 30 characters'),
+    surname: Yup.string()
+        .required('Surname is required')
+        .min(10, 'Surname must be at least 10 characters')
+        .max(30, 'Surname must not exceed 30 characters'),
+    userName: Yup.string()
+        .required('User name is required')
+        .min(10, 'User name must be at least 10 characters')
+        .max(30, 'User name must not exceed 30 characters'),
+    email: Yup.string()
+        .required('Email is required')
+        .email('Email is invalid'),
+    address: Yup.string()
+        .required('Address is required')
+        .min(0, 'Address must be at least 0 characters')
+        .max(30, 'Address must not exceed 30 characters'),
+    password: Yup.string()
+        .required('Password is required')
+        .min(6, 'Password must be at least 6 characters')
+        .max(40, 'Password must not exceed 40 characters'),
+    confirmPassword: Yup.string()
+        .required('Confirm Password is required')
+        .oneOf([Yup.ref('password'), null], 'Confirm Password does not match'),
+    cvuMP: Yup.string()
+        .required('CVU MercadoPago is required')
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/JeremiasFuentes/unq-desapp-fuentes-diaz/pullRequest?prid=8198099)
```suggestion
        .required("CVU MercadoPago is required")
```